code/stl/Complex.hpp

```diff
@@ -3,72 +3,51 @@
 template <typename T=float>
 class Complex_t {
 public:
-    Complex_t(T r, T i);
-    Complex_t();
+    Complex_t() = default;
+    Complex_t(T r, T i) : m_r(r), m_i(i) {}
 
     T real() const {return m_r;};
     T imaginary() const {return m_i;};
 
-    Complex_t operator+(const Complex_t& other) const;
-    Complex_t operator-(const Complex_t& other) const;
-    Complex_t operator*(const Complex_t& other) const;
-    Complex_t operator*(const T factor) const;
-    Complex_t operator/(const T dividend) const;
-    Complex_t& operator+=(const Complex_t& other);
-    bool operator<(const Complex_t& a) const;
-private:
-    T m_r{}, m_i{};
-};
-
-typedef Complex_t<> Complex;
+    Complex_t& operator+=(const Complex_t& other) {
+        m_r += other.m_r;
+        m_i += other.m_i;
+        return *this;
+    }
 
-template <typename T>
-Complex_t<T>::Complex_t() {}
+    friend Complex_t operator+(Complex_t a, const Complex_t& b) {
+        return a += b;
+    }
 
-template <typename T>
-Complex_t<T>::Complex_t(T r, T i) : m_r(r), m_i(i) {}
+    friend Complex_t operator-(const Complex_t& a, const Complex_t& b) {
+        return Complex_t(a.m_r - b.m_r, a.m_i - b.m_i);
+    }
```

Comment:
Not your code initially, but why do we provide `+=` and not `-=`, etc?
Maybe this is unnecessary, indeed, but if you want to reuse it somewhere else, one could imagine to implement `{+,-,*,/}=` and express all of {+,-,*,/} in terms of those.

Reply:
Are hidden friends in the slides?
Hint: Yes, to explain `swap`, but we don't discuss them. 🙂 
Issue coming ...

Reply:
I added the missing operators.

And thx for opening the issue to discuss hidden friends.